### ISSUE TITLE
[$250] Remove MoneyRequestParticipantsPage.js and copy any changes since Nov 27 into IOURequestStepParticipants.js

### ISSUE THREAD

**Issue by tgolen** [Expensify/App#34616]
This is a part of https://github.com/Expensify/App/issues/29107. You can look at that issue for more context behind the cleanup process.

# Problem
The app has two redundant components:

**Old Component**: `MoneyRequestParticipantsPage`
**New Component** `IOURequestStepParticipants`

# Solution
Following the examples ([example 1](https://github.com/Expensify/App/pull/33646), [example 2](https://github.com/Expensify/App/pull/32764)), the Old Component needs to be completely removed from the codebase

1. Look at the history of the Old Component
1. If there are any changes since Nov 27, 2023 which have not been added to the New Component, copy those changes
1. Replace all uses of the Old Component with the New Component
2. Remove all traces of Old Component
3. Be sure to update all routes and navigation to use the new `:action` param (instead of being hard-coded with `"create"`)
4. Update any logic like `isEditing` to use the new action param from the route
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~015fc3b79f31f2289a</li>
        <li>Upwork Job ID: 1747632960650227712</li>
        <li>Last Price Increase: 2024-01-24</li>
<li>Automatic offers: </li>
<ul>
<li>abdulrahuman5196 | Reviewer | 28131654</li>
<li>Krishna2323 | Contributor | 28131655</li>
</ul></li>
    </ul>
</details>

**Comment by njmulsqb** [comment]
I would love to take this!

**Comment by MahmudjonToraqulov** [comment]
I would love to take this!

**Comment by isabelastisser** [comment]
@0xmiroslav, can you please review the proposal above? Thanks!

**Comment by mallenexpensify** [comment]
@abdulrahuman5196 reassigning, please take over as C+. If you don't have bandwidth, unassign yourself. Thanks

**Comment by brunovjk** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
This belongs to [Wave 5 cleanup](https://github.com/Expensify/App/issues/29107)
Phase 2 is all about removing the original components and any of the duplicated efforts.

### What is the root cause of that problem?
Replace `MoneyRequestParticipantsPage` to `IOURequestStepParticipants`. **This new component will handle both the create and edit flow based on `action` param**.

### What changes do you think we should make in order to solve the problem?

> Look at the history of the Old Component
> If there are any changes since Nov 27, 2023 which have not been added to the New Component, copy those changes

1. Checked the [there is no commits](https://github.com/Expensify/App/commits/main/src/pages/iou/MoneyRequestParticipantsPage.js?since=2023-11-27&until=2024-01-23) made to the `MoneyRequestParticipantsPage` component since 27Nov. 
- This way there will be **no need to arrange any changes** from `MoneyRequestParticipantsPage` to be transferred to `IOURequestStepParticipants`

> Replace all uses of the Old Component with the New Component
> Remove all traces of Old Component

3. There is no use of the component other than `ModalStackNavigators`. We must remove the `MoneyRequestParticipantsPage` component along with:
https://github.com/Expensify/App/blob/f56f1809846822f35d3bca8762a302086df1b774/src/libs/Navigation/AppNavigator/ModalStackNavigators.tsx#L95

4. The new component uses `STEP_PARTICIPANTS` and has it own [`linkingConfig`](https://github.com/Expensify/App/blob/278d04e0689d6f012354d89ec19c640a1cd9dc54/src/libs/Navigation/linkingConfig.ts#L403), [`SCREENS`](https://github.com/Expensify/App/blob/f56f1809846822f35d3bca8762a302086df1b774/src/SCREENS.ts#L132) and [`ROUTES`](https://github.com/Expensify/App/blob/f56f1809846822f35d3bca8762a302086df1b774/src/ROUTES.ts#L361-L365) already configured, we must **remove** `PARTICIPANTS` from [`linkingConfig`](https://github.com/Expensify/App/blob/278d04e0689d6f012354d89ec19c640a1cd9dc54/src/libs/Navigation/linkingConfig.ts#L428), [`SCREENS`](https://github.com/Expensify/App/blob/f56f1809846822f35d3bca8762a302086df1b774/src/SCREENS.ts#L140) and [`ROUTES`](https://github.com/Expensify/App/blob/f56f1809846822f35d3bca8762a302086df1b774/src/ROUTES.ts#L261-L264)

> Be sure to update all routes and navigation to use the new :action param (instead of being hard-coded with "create")

5. Update [route and getRoute](https://github.com/Expensify/App/blob/0706c8c1a8914fa5ef4a7a05630e793ff691fd0a/src/ROUTES.ts#L346-L350) to use the new :action param.
    ```
        MONEY_REQUEST_STEP_PARTICIPANTS: {
            route: ':action/:iouType/participants/:transactionID/:reportID?',
            getRoute: (action: ValueOf<typeof CONST.IOU.ACTION>, iouType: ValueOf<typeof CONST.IOU.TYPE>, transactionID: string, reportID: string, backTo = '') =>
                getUrlWithBackToParam(`${action}/${iouType}/participants/${transactionID}/${reportID}`, backTo),
        },
    ```

6. Update [Type navigation](https://github.com/Expensify/App/blob/0706c8c1a8914fa5ef4a7a05630e793ff691fd0a/src/libs/Navigation/types.ts#L184-L187) according to new `route and getRoute`.
    ```
        [SCREENS.MONEY_REQUEST.STEP_PARTICIPANTS]: {
            action: string;
            ...
        };
    ```
7. Add `CONST.IOU.ACTION.CREATE` at several places (IOURequestStepAmount.js, IOURequestStepDistance.js, IOURequestStepTaxAmountPage.js, IOURequestStepConfirmation.js, IOURequestStepTaxAmountPage.js, IOURequestStepScan) where we are using we are using `MONEY_REQUEST_STEP_PARTICIPANTS.getRoute`
    ```
    Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_PARTICIPANTS.getRoute(CONST.IOU.ACTION.CREATE, iouType, transaction.transactionID, reportID, Navigation.getActiveRouteWithoutParams()));
    ```
    
> Update any logic like isEditing to use the new action param from the route

- `MoneyRequestParticipantsPage` is only used for the create flow. **So we don’t need to use `action` param**, is not utilized as a route. Consequently, the outlined steps are unnecessary.

### What alternative solutions did you explore? (Optional)
N/A

**Comment by abdulrahuman5196** [comment]
Yes. Will review proposals today.

**Comment by abdulrahuman5196** [comment]
Reviewing now

**Comment by abdulrahuman5196** [comment]
This is a straightforward task with clear requirements in OP. So approving the first proposal with some information regarding the changes.
@Krishna2323 's proposal here https://github.com/Expensify/App/issues/34616#issuecomment-1894614055 looks good and works well.

🎀 👀 🎀
C+ Reviewed

**Comment by abdulrahuman5196** [comment]
Waiting on @Krishna2323 for PR

**Comment by abdulrahuman5196** [comment]
Got it. I agree we could wait for the PR, to avoid cross conflicts given this is LOW priority. 

@Krishna2323 since you are owner of the PR as well, kindly do update us once the PR is ready for review.

**Comment by isabelastisser** [comment]
@Krishna2323 @abdulrahuman5196 what's the next step here?

**Comment by DylanDylann** [comment]
@Krishna2323 Could you speed-up this issue to unblock other issues

**Comment by isabelastisser** [comment]
Hey @abdulrahuman5196, can you please provide an update? Thanks!

**Comment by mountiny** [comment]
Asked in slack too https://expensify.slack.com/archives/C02NK2DQWUX/p1711752832574839

**Comment by isabelastisser** [comment]
@Krishna2323 and @abdulrahuman5196, please accept the offers in Upwork and I will process the payments. Thanks!

**Comment by abdulrahuman5196** [comment]
@isabelastisser Accepted the offer.
